drivers/flash/flash_hp_ra.c

```diff
@@ -0,0 +1,392 @@
+		flash_ra_layout[0].pages_size = FLASH_HP_CF_BLOCK_8KB_SIZE;
+		flash_ra_layout[1].pages_count =
+			(FLASH_HP_CF_BLOCK_32KB_LINEAR_END - FLASH_HP_CF_BLOCK_32KB_LINEAR_START) +
+			1;
+		flash_ra_layout[1].pages_size = FLASH_HP_CF_BLOCK_32KB_SIZE;
+
+		*layout_size = 2;
+#endif
+	}
+
+	*layout = flash_ra_layout;
+}
+
+static const struct flash_parameters *flash_ra_get_parameters(const struct device *dev)
+{
+	const struct flash_hp_ra_config *config = dev->config;
+
+	return &config->flash_ra_parameters;
+}
+
+static struct flash_hp_ra_controller flash_hp_ra_controller = {
+	.fsp_config = {
+		.data_flash_bgo = true,
+		.p_callback = bgo_callback,
+		.p_context = NULL,
+		.irq = (IRQn_Type)DT_INST_IRQ_BY_NAME(0, frdyi, irq),
+		.err_irq = (IRQn_Type)DT_INST_IRQ_BY_NAME(0, fiferr, irq),
+		.err_ipl = DT_INST_IRQ_BY_NAME(0, fiferr, priority),
+		.ipl = DT_INST_IRQ_BY_NAME(0, frdyi, priority),
+	}};
+
+#ifdef CONFIG_FLASH_EX_OP_ENABLED
+static int flash_ra_ex_op(const struct device *dev, uint16_t code, const uintptr_t in, void *out)
+{
+	int err = -ENOTSUP;
+
+	switch (code) {
+#if defined(CONFIG_FLASH_RA_WRITE_PROTECT)
+	case FLASH_RA_EX_OP_WRITE_PROTECT:
+		err = flash_ra_ex_op_write_protect(dev, in, out);
+		break;
+#endif /* CONFIG_FLASH_RA_WRITE_PROTECT */
+
+	default:
+		break;
+	}
+
+	return err;
+}
+#endif
+
+static int flash_ra_init(const struct device *dev)
+{
+	const struct device *dev_ctrl = DEVICE_DT_INST_GET(0);
+	const struct flash_hp_ra_config *cfg = dev->config;
+	struct flash_hp_ra_data *flash_data = dev->data;
+
+	if (!device_is_ready(dev_ctrl)) {
+		return -ENODEV;
+	}
+
+	if (cfg->flash_ra_parameters.write_block_size == UNIT_CODE_FLASH_PROGRAMMING) {
+		flash_data->FlashRegion = CODE_FLASH;
+	} else {
+		flash_data->FlashRegion = DATA_FLASH;
+	}
+
+	flash_data->controller = dev_ctrl->data;
+
+	return 0;
+}
+
+static void flash_controller_ra_irq_config_func(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	R_ICU->IELSR[DT_IRQ_BY_NAME(DT_DRV_INST(0), frdyi, irq)] = ELC_EVENT_FCU_FRDYI;
+	R_ICU->IELSR[DT_IRQ_BY_NAME(DT_DRV_INST(0), fiferr, irq)] = ELC_EVENT_FCU_FIFERR;
+
+	IRQ_CONNECT(DT_IRQ_BY_NAME(DT_DRV_INST(0), frdyi, irq),
+		    DT_IRQ_BY_NAME(DT_DRV_INST(0), frdyi, priority), fcu_frdyi_isr,
+		    DEVICE_DT_INST_GET(0), 0);
+	IRQ_CONNECT(DT_IRQ_BY_NAME(DT_DRV_INST(0), fiferr, irq),
+		    DT_IRQ_BY_NAME(DT_DRV_INST(0), fiferr, priority), fcu_fiferr_isr,
+		    DEVICE_DT_INST_GET(0), 0);
+
+	irq_enable(DT_INST_IRQ_BY_NAME(0, frdyi, irq));
+	irq_enable(DT_INST_IRQ_BY_NAME(0, fiferr, irq));
+}
+
+static int flash_controller_ra_init(const struct device *dev)
+{
+	fsp_err_t err = FSP_SUCCESS;
+	const struct flash_hp_ra_controller_config *cfg = dev->config;
+	struct flash_hp_ra_controller *data = dev->data;
+
+	cfg->irq_config(dev);
+
+	err = R_FLASH_HP_Open(&data->flash_ctrl, &data->fsp_config);
+	LOG_DBG("flash: open error=%d", err);
+
+	if (err != FSP_SUCCESS) {
+		return -EIO;
```

Comment:
You can just move the LOG_DBG into if, you probably only want to see this when the failure appears. Also, the `err` is of `fsp_err_t`, so you are probably missing cast to `int` in the LOG_DBG.

Reply:
I moved it into if, thanks.